engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailProcessor.java

```diff
@@ -0,0 +1,181 @@
+package com.redhat.cloud.notifications.processors.email;
+
+import com.redhat.cloud.event.core.v1.Recipients;
+import com.redhat.cloud.notifications.db.repositories.EmailAggregationRepository;
+import com.redhat.cloud.notifications.db.repositories.EndpointRepository;
+import com.redhat.cloud.notifications.db.repositories.NotificationHistoryRepository;
+import com.redhat.cloud.notifications.db.repositories.TemplateRepository;
+import com.redhat.cloud.notifications.events.EventWrapper;
+import com.redhat.cloud.notifications.ingress.Action;
+import com.redhat.cloud.notifications.models.EmailAggregation;
+import com.redhat.cloud.notifications.models.EmailSubscriptionType;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.EndpointType;
+import com.redhat.cloud.notifications.models.Event;
+import com.redhat.cloud.notifications.models.EventType;
+import com.redhat.cloud.notifications.models.InstantEmailTemplate;
+import com.redhat.cloud.notifications.models.NotificationHistory;
+import com.redhat.cloud.notifications.models.NotificationStatus;
+import com.redhat.cloud.notifications.models.NotificationsConsoleCloudEvent;
+import com.redhat.cloud.notifications.processors.ConnectorSender;
+import com.redhat.cloud.notifications.processors.SystemEndpointTypeProcessor;
+import com.redhat.cloud.notifications.processors.email.connector.dto.EmailNotification;
+import com.redhat.cloud.notifications.processors.email.connector.dto.RecipientSettings;
+import com.redhat.cloud.notifications.transformers.BaseTransformer;
+import io.quarkus.logging.Log;
+import io.vertx.core.json.JsonObject;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class EmailProcessor extends SystemEndpointTypeProcessor {
+    @Inject
+    BaseTransformer baseTransformer;
+
+    @Inject
+    ConnectorSender connectorSender;
+
+    @Inject
+    EndpointRepository endpointRepository;
+
+    @Inject
+    EmailAggregationRepository emailAggregationRepository;
+
+    @Inject
+    NotificationHistoryRepository notificationHistoryRepository;
+
+    @Inject
+    TemplateRepository templateRepository;
+
+    /**
+     * The non-instant subscription types that we support.
+     */
+    private static final List<EmailSubscriptionType> NON_INSTANT_SUBSCRIPTION_TYPES =
+        Arrays
+            .stream(EmailSubscriptionType.values())
+            .filter(emailSubscriptionType -> emailSubscriptionType != EmailSubscriptionType.INSTANT)
+            .collect(Collectors.toList());
+
+
+    @Override
+    public void process(final Event event, final List<Endpoint> endpoints) {
+        if (endpoints == null || endpoints.isEmpty()) {
+            return;
+        }
+
```

Comment:
We'll need something similar to the following code to prevent one failure from interrupting the processing of the list of endpoints:
https://github.com/RedHatInsights/notifications-backend/blob/e506ceaa0ce8b10015d07fca028e6cad16ccbda9/engine/src/main/java/com/redhat/cloud/notifications/processors/eventing/EventingProcessor.java#L53-L61

Reply:
Done! Although we can spare the loop, since we only need the endpoints to extract the recipient settings from them.